caffe2/core/c10_operator.h

```diff
@@ -149,72 +160,47 @@ inline c10::FunctionSchema make_function_schema_for_c10(const char* OperatorName
-          OperatorClass,                                                       \
-          at::DeviceType::CUDA>>()                                             \
-      .dispatchKey(CUDATensorId());                                            \
+#define C10_REGISTER_CAFFE2_OPERATOR_CUDA(OperatorName, OperatorClass)        \
+  namespace c10 {                                                             \
+  C10_REGISTER_KERNEL(caffe2::_c10_ops::OperatorName) /*.withCache<Cache>()*/ \
+      .kernel<&caffe2::detail::call_caffe2_op_from_c10<                       \
+          ::caffe2::_c10_ops::OperatorName,                                   \
+          OperatorClass>>()                                                   \
+      .dispatchKey(CUDATensorId());                                           \
   }
 
 // You should never manually call the C10_REGISTER_CAFFE2_OPERATOR_HIP macro.
 // The C10_REGISTER_CAFFE2_OPERATOR_CUDA macro from above will be automatically
 // rewritten to C10_REGISTER_CAFFE2_OPERATOR_HIP by hipify.
-#define C10_REGISTER_CAFFE2_OPERATOR_HIP(OperatorName, OperatorClass)          \
-  /* Store the c10 operator handle so call_caffe2_op_from_c10 can access it */ \
-  namespace caffe2 {                                                           \
-  namespace detail {                                                           \
-  template <>                                                                  \
-  const c10::OperatorHandle& c10_op_handle_for_c2_op<OperatorClass>() {        \
-    return caffe2::_c10_ops::OperatorName();                                   \
-  }                                                                            \
-  }                                                                            \
-  }                                                                            \
-  namespace c10 {                                                              \
-  C10_REGISTER_KERNEL(caffe2::_c10_ops::OperatorName) /*.withCache<Cache>()*/  \
-      .kernel<&caffe2::detail::call_caffe2_op_from_c10<                        \
-          OperatorClass,                                                       \
-          at::DeviceType::HIP>>()                                              \
-      .dispatchKey(CUDATensorId());                                            \
+#define C10_REGISTER_CAFFE2_OPERATOR_HIP(OperatorName, OperatorClass)         \
+  namespace c10 {                                                             \
+  C10_REGISTER_KERNEL(caffe2::_c10_ops::OperatorName) /*.withCache<Cache>()*/ \
+      .kernel<&caffe2::detail::call_caffe2_op_from_c10<                       \
+          ::caffe2::_c10_ops::OperatorName,                                   \
+          OperatorClass>>()                                                   \
+      .dispatchKey(CUDATensorId());                                           \
```

Comment:
I don't understand, if this is a HIP operator, why is the tensor id here CUDA

Reply:
ouch good catch. This bug was introduced with my last change to this PR because I redid the copy&paste from the cuda implementation above. Fixed.